src/CommonAPI/IniFileReader.cpp

```diff
@@ -29,65 +29,66 @@ IniFileReader::Section::getValue(const std::string &_key) const {
 bool
 IniFileReader::load(const std::string &_path) {
     std::ifstream configStream(_path);
-    if (configStream.is_open()) {
-        //COMMONAPI_INFO("Loading ini file from ", _path);
+    if (!configStream.is_open()) {
+        COMMONAPI_ERROR("Failed to load ini file: ", _path);
+        return false;
+    }
 
-        int lineCounter(0);
-        std::string currentSectionName;
-        std::shared_ptr<Section> currentSection;
+    int lineCounter(0);
+    std::string currentSectionName;
+    std::shared_ptr<Section> currentSection;
 
-        while (!configStream.eof()) {
-            std::string line;
-            std::getline(configStream, line);
-            lineCounter++;
+    while (!configStream.eof()) {
+        std::string line;
+        std::getline(configStream, line);
+        lineCounter++;
 
-            trim(line);
+        trim(line);
 
-            std::size_t start = line.find('[');
-            if (start == 0) {
-                std::size_t end = line.find(']');
-                if (end != line.npos) {
-                    currentSectionName = line.substr(++start, --end);
-                    if (sections_.end() == sections_.find(currentSectionName)) {
-                        currentSection = std::make_shared<Section>();
-                        if (currentSection) {
-                            sections_[currentSectionName] = currentSection;
-                        }
-                    } else {
-                        COMMONAPI_ERROR("Double definition of section \'",
-                                        currentSectionName,
-                                        "\' ignoring definition (line ",
-                                        lineCounter,
-                                        ")");
-                        currentSection = nullptr;
+        std::size_t start = line.find('[');
+        if (start == 0) {
+            std::size_t end = line.find(']');
+            if (end != line.npos) {
+                currentSectionName = line.substr(++start, --end);
+                if (sections_.end() == sections_.find(currentSectionName)) {
+                    currentSection = std::make_shared<Section>();
+                    if (currentSection) {
+                        sections_[currentSectionName] = currentSection;
                     }
                 } else {
-                    COMMONAPI_ERROR("Missing \']\' in section definition (line ",
-                                    lineCounter, ")");
+                    COMMONAPI_ERROR("Double definition of section \'",
```

Comment:
Seems like there should be more consideration regarding the return value.  What's the appropriate return value for these other COMMONAPI_ERROR cases?